kornia/filters/kernels.py

```diff
@@ -89,7 +89,7 @@ def gaussian(
 
     mean = float(window_size // 2) if mean is None else mean
     if isinstance(mean, float):
-        mean = tensor([[mean]], device=device, dtype=dtype)
+        mean = tensor([[mean]], device=sigma.device, dtype=sigma.dtype)
```

Comment:
this function has the device/dtype as a parameter ... What I believe to be ideal here is, if these parameters have not been set, to use the same ones as for sigma
```suggestion
        mean = tensor([[mean]], device=device or sigma.device, dtype=dtype or sigma.dtype)
```

Reply:
> this function has the device/dtype as a parameter ... What I believe to be ideal here is, if these parameters have not been set, to use the same ones as for sigma

This doesn't work, because sometimes they are set, however according to the common sense and doctoring, the sigma device should take a priority